sign/ed25519/ed25519_test.go

```diff
@@ -188,3 +243,30 @@ func Example_ed25519() {
 	fmt.Println(ok)
 	// Output: true
 }
+
+func Example_ed25519Ph() {
+	// import "github.com/cloudflare/circl/sign/ed25519"
+
+	// Generating Alice's key pair
+	keys, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		panic("error on generating keys")
+	}
+
+	// Alice signs a message.
+	message := []byte("A message to be signed")
+	opts := crypto.SHA512
+	h := sha512.New()
```

Comment:
This is not the right example, as the prehashed mode avoids the user hashes the message.

Reply:
Oh, really? It was my understanding that it expected the input to be  hashed using SHA-512 before signing with Ed25519: "For Ed25519ph, phflag=1 and PH is SHA512 instead.  That is, the input is hashed using SHA-512 before signing with Ed25519". But maybe it is not the user that manually hashes it.. is that what you mean? The example can work as well with out the hash but I thought it made it clearer.. what do you think?

Reply:
> But maybe it is not the user that manually hashes it.. is that what you mean?

yes

Reply:
My bad: the routine actually fails if the message is not hashed (as it also happens on the RFC test cases). I'll add a note that says 'the message is hashed'.

Reply:
I believe there is a misconception, I added a comment ([here](https://github.com/golang/go/issues/31804#issuecomment-653405864)).

Reply:
I see what you mean now. Mmm.. hashing internally might be the right way to go for two reasons:

1. The function can't know if the message was hashed or not, and there is no way of knowing
2. The user should not need to manually hash, as it will create confusion for them

I also read 'the input is hashed using SHA-512 before signing with Ed25519.' on those terms now.

Alright, I'll change the functions ;)

Reply:
Should be fixed now.